src/targets/upm.ts

```diff
@@ -0,0 +1,141 @@
+    this.github = getGithubClient();
+  }
+
+  /**
+  * Extracts upm target options from the environment.
+  */
+  protected getUpmConfig(): UpmTargetConfig {
+    return {
+      releaseTarget: (this.config.releaseTarget === undefined)
+        ? DEFAULT_RELEASE_REMOTE
+        : new GithubRemote('getsentry', this.config.releaseTarget)
+    };
+  }
+
+  /**
+   * Fetches the artifact for the provided revision.
+   *
+   * @param revision Git commit SHA for the to be published artefact.
+   * @returns The requested artefact, undefined of no or multiple artefacts
+   * have been found.
+   */
+  public async fetchArtefact(revision: string): Promise<RemoteArtifact> {
+    const packageFiles = await this.getArtifactsForRevision(revision);
+    if (packageFiles.length == 0){
+      reportError('Cannot publish UPM: No release artefact found.')
+      return undefined;
+    } else if(packageFiles.length > 1) {
+      reportError('Cannot publish UPM: Too many release artefacts found:')
+      packageFiles.forEach(function (packageFile) {
+        logger.info(`${packageFile.filename}`);
+      });
+      return undefined;
+    }
+    return packageFiles[0];
+  }
+
+  /**
+   * Performs a release to upm
+   *
+   * @param version New version to be released
+   * @param revision Git commit SHA to be published
+   */
+  public async publish(version: string, revision: string): Promise<any> {
+    logger.info(`Target "${this.name}": publishing version "${version}"...`);
+
+    logger.info("Fetching artefact.");
+    const packageFile = await this.fetchArtefact(revision);
+    if (packageFile === undefined) {
+      return undefined;
+    }
+
+    logger.info(`Artefact: "${packageFile.filename}"`);
+    const artefactPath = await this.artifactProvider.downloadArtifact(
+      packageFile
+    );
+
+    logger.info(`Target release repository: `+
+      `"${this.upmConfig.releaseTarget.getRemoteString}"`);
+    const remote = this.upmConfig.releaseTarget;
+    const username = await getAuthUsername(this.github);
+    remote.setAuth(username, getGithubApiToken());
+
+    await withTempDir(
+      async directory => {
+        const git = simpleGit(directory);
+        logger.info(`Cloning ${remote.getRemoteString()} to ${directory}...`);
+        await git.clone(remote.getRemoteStringWithAuth(), directory);
+
+        logger.info('Clearing the repository.')
+        await git.raw(['rm', '-r', '--', '.']);
+
+        logger.info(`Extracting "${packageFile.filename}".`)
+        await extractZipArchive(artefactPath, directory)
+
+        logger.info('Adding files to repository.')
+        await git.add('.');
+
+        if (isDryRun) {
+          logger.info(`[dry-run]: git commit -m "release ${version}"`);
+          logger.info(`[dry-run]: git tag ${version}`);
+          logger.info('[dry-run]: git push origin main --tags');
+        } else {
+          await git.commit(`release ${version}`);
+          await git.tag(version)
```

Comment:
Would be nice to also create a GitHub release and add the release notes.
Basically duplicating what the `github` target does but on this other repo we're pushing to.